seahorse-workflow-executor/deeplang/src/main/scala/ai/deepsense/deeplang/doperables/dataframe/DataFrame.scala

```diff
@@ -35,7 +35,7 @@ import ai.deepsense.deeplang.{DOperable, ExecutionContext}
  * @param schema Schema of the DataFrame. Usually it is schema of sparkDataFrame,
  *               but for inference, DataFrame may be not set but schema is known.
  */
-case class DataFrame private[dataframe] (
+case class DataFrame protected[dataframe] (
```

Comment:
@jaroslaw-osmanski Assuming this is in response to my email, do you also want to drop the `[dataframe]` part?

Reply:
Protected[dataframe] allows using constructor in subclass, so not at this time. Unless you have need for that.

Reply:
@jaroslaw-osmanski We have a couple of scenarios where it's helpful to be able to subclass the deeplang DataFrame.  Because it's private, we've hacked some workarounds by calling the default constructor and overriding the fields, but I'm wondering if  you guys would consider changing the constructor access restrictions; either to public, or perhaps protected? With the DOperable registration we're able to have nodes return DataFrames with certain guarantees placed on them and have them still work with normal nodes, as well as enable special nodes to have more restrictive inputs.

So, what are your thoughts about opening up DataFrame? Concerns?

Reply:
We've made it protected in this PR and now each subclass (even from SDK package) is able to use DataFrame. This should work:
```
package sdk.notdataframe {
class ChildDataFrame(val sparkDataFrame, val optionalSchema) extends DataFrame(sparkDataFrame, optionalSchema);
} 
```

protected[dataframe] would prohibit:
```package sdk.notdataframe {
class Wild {
  val b = new DataFrame(sparkDataFrame, optionalSchema);
}
}
```
I don't want to make DataFrame constructor public as apply from DataFrame object should be used. ChildDataFrame should cover your scenario with subclassed DataFrame, wouldn't it? Maybe protected[dataframe] doesn't work as I've tested it (ie. it doesn't allow using constructor outside *.deeplang.*.dataframe package. Then I'll change that to public.

Reply:
Interesting.... I had thought `protected[dataframe]` would have prohibited

```scala
package sdk.notdataframe {
class ChildDataFrame(val sparkDataFrame, val optionalSchema) extends DataFrame(sparkDataFrame, optionalSchema);
}
```

which is our use case.... so as long as that works, we're fine.

Reply:
@metasim Can you confirm that it works?

Reply:
@jaroslaw-osmanski I can confirm that this does what we need. My mental model of `protected[dataframe]` was wrong.... good to learn something new!